### PR TITLE
Replaced the actionBar with custom Toolbars

### LINKED ISSUE #52: Replace the actionbar with a toolbar
This is self explanatory. Replace the actionbar with a toolbar using the support library.

### PR DESCRIPTION
**Pull request by octacode**
Fixes #52.

- Made two toolbars.(toolbar.xml and toolbar_transparent.xml).
- Now all the activities extend AppCompatActivity instead of Activity except Settings.
- Made a custom toolbar with an image and textView for the NavigationDrawerActivity.

### CODE REVIEW

**Comment by octacode** [comment]
I have resolved the merge conflicts here.

**Comment by megabitdragon** [comment]
While this fixes the conflict it breaks functionality. To test, open a picture and try to go back using the arrow in the navigation toolbar. In my tests this is not working.

**Comment by octacode** [comment]
@megabitdragon I wasn't able to replicate the issue in my Moto phone. Can you please provide the logs of error.

**Comment by megabitdragon** [comment]
screen
There is no error in the log. It is a functionality issue.
This is how it works before pr #67 
screen_good

**Comment by csoni111** [comment]
@octacode The problem is occurring because in my pr, I have used `ActionBar` instead of `SupportActionBar`. So will have to update my code also for resolving this issue. 
Follow these steps and your pr will be good to go:

- First pull the latest master from amahi/android to your fork
- Rebase your `toolbar_updated` branch on top of master
- Solve the merge conflicts like this (in *activity_server_file_image.xml*)
    - Instead of `ViewPager` -> use `ClickableViewPager`
    - In `RelativeLayout` add this -> `android:fitsSystemWindows="true"`
    - Move the `toolbar include tag` below the `ClickableViewPager` , so that it appears on top of Image
    - Give toolbar a *light black translucent* background
- Then in `util/FullScreenHelper.java` change `import android.app.ActionBar;` -> `import android.support.v7.app.ActionBar;`

After this everything should work fine.

Cheers!

File: src/main/res/layout/activity_server_files.xml [hunk 1]
```diff
@@ -19,7 +19,12 @@
   ~ along with Amahi. If not, see <http ://www.gnu.org/licenses/>.
   -->
 
-<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<RelativeLayout
```
Comment (csoni111) [inline src/main/res/layout/activity_server_files.xml]: Keep this as `LinearLayout` with orientation vertical, this way you will not have to add margins to all the ViewAnimators present in the layouts of all the fragments.

File: src/main/res/layout/fragment_server_files_metadata.xml [hunk 2]
```diff
@@ -20,6 +20,7 @@
 
 <ViewAnimator xmlns:android="http://schemas.android.com/apk/res/android"
 	android:id="@+id/animator"
+	android:layout_marginTop="?attr/actionBarSize"
```
Comment (csoni111) [inline src/main/res/layout/fragment_server_files_metadata.xml]: Same things

File: src/main/res/layout/fragment_server_shares.xml [hunk 3]
```diff
@@ -20,6 +20,7 @@
 
 <ViewAnimator xmlns:android="http://schemas.android.com/apk/res/android"
 	android:id="@+id/animator"
+	android:layout_marginTop="?attr/actionBarSize"
```
Comment (csoni111) [inline src/main/res/layout/fragment_server_shares.xml]: Same thing

File: src/main/res/layout/activity_navigation.xml [hunk 4]
```diff
@@ -19,21 +19,47 @@
...
+			android:id="@+id/toolbar"
+			android:minHeight="?attr/actionBarSize"
+			app:theme="@style/ThemeOverlay.AppCompat.ActionBar">
+
+			<ImageView
+				android:layout_width="50dp"
+				android:layout_height="50dp"
+				android:src="@drawable/ic_app_logo"/>
+
+			<TextView
```
Comment (csoni111) [inline src/main/res/layout/activity_navigation.xml]: There's no need for adding `TextView` and `ImageView` inside Toolbar, to display the title and app icon you may use this:
`getSupportActionBar().setDisplayShowHomeEnabled(true);`
Along with
`getSupportActionBar().setIcon(R.drawable.ic_launcher);`
`getSupportActionBar().setTitle("Title");`

File: src/main/res/values-v21/themes.xml [hunk 5]
```diff
@@ -1,26 +1,27 @@
...
+	<style name="Theme.Amahi.Fullscreen" parent="@style/Theme.AppCompat.NoActionBar">
 		<item name="android:windowActionBarOverlay">true</item>
 		<item name="android:windowBackground">@android:color/black</item>
 		<item name="android:windowContentOverlay">@null</item>
 	</style>
 
+	<style name="Theme.Amahi.Settings" parent="@android:style/Theme.Material">
+		<item name="android:actionBarStyle">@style/Theme.Amahi.ActionBar</item>
+		<item name="android:windowBackground">@color/background_primary</item>
+		<item name="android:colorAccent">@color/accent_material_dark</item>
```
Comment (csoni111) [inline src/main/res/values-v21/themes.xml]: Android Studio gives a warning here: `The resource @color/accent_material_dark is marked as private in com.android.support:appcompat-v7`.
Please fix this.

File: src/main/res/values-v21/themes.xml [hunk 6]
```diff
@@ -1,26 +1,27 @@
...
+		<item name="android:textColorPrimaryInverse">@color/primary_text_material_light</item>
+	</style>
+
 	<style name="Theme.Amahi.ActionBar" parent="@android:style/Widget.Material.ActionBar.Solid">
 		<item name="android:background">@drawable/bg_action_bar</item>
-		<item name="android:displayOptions">showHome|homeAsUp|showTitle</item>
+		<item name="android:displayOptions">homeAsUp|showTitle</item>
```
Comment (csoni111) [inline src/main/res/values-v21/themes.xml]: Why have you removed `showHome` flag? This hides the app icon from the toolbar in Settings activity.

File: src/main/java/org/amahi/anywhere/activity/NavigationActivity.java [hunk 7]
```diff
@@ -81,20 +84,25 @@ protected void onCreate(Bundle savedInstanceState) {
...
+		getDrawer().setScrimColor(getResources().getColor(android.R.color.transparent));
 	}
 
 	private void setUpInjections() {
 		AmahiApplication.from(this).inject(this);
 	}
 
-	private void setUpHomeNavigation() {
-		ActionBar actionBar = getActionBar();
+	private TextView getToolbarTitle(){return (TextView)findViewById(R.id.toolbar_title);}
```
Comment (csoni111) [inline src/main/java/org/amahi/anywhere/activity/NavigationActivity.java]: This is not required, reason mentioned in its respective layout file.

File: src/main/java/org/amahi/anywhere/activity/NavigationActivity.java [hunk 8]
```diff
@@ -139,7 +147,7 @@ public void onDrawerOpened(View drawer) {
 
 	private void setUpTitle(String title) {
 		if (isNavigationDrawerAvailable()) {
-			getActionBar().setTitle(title);
+			getToolbarTitle().setText(title);
```
Comment (csoni111) [inline src/main/java/org/amahi/anywhere/activity/NavigationActivity.java]: Simply replace it with `getSupportActionBar().setTitle(title);`

**Comment by csoni111** [comment]
@octacode @megabitdragon  After all this I just don't understand the need for updating the app to using `Custom Toolbars` ? Ofcourse updating to `AppCompatActivity` is a necessary change but toolbar/actionbar can used directly by specifying Theme parent as one with ActionBar.

**Comment by octacode** [comment]
@csoni111 Thanks for the review. I will incorporate the changes.

File: src/main/res/values-v21/themes.xml (hunk 6 above)
Comment (octacode) [inline src/main/res/values-v21/themes.xml]: Well I have created a new theme for Settings Activity.

File: src/main/res/values-v21/themes.xml (hunk 5 above)
Comment (octacode) [inline src/main/res/values-v21/themes.xml]: Ya, sure will do

**Comment by csoni111** [comment]
Why does your commit [94b83a7](https://github.com/amahi/android/pull/67/commits/94b83a7ee4c09076437b1e9157ce4f15e21ea8b4) contains the code that I added in my last PR? I think instead of rebasing you merged the branches, creating duplicate commits!

Please fix this.

**Comment by octacode** [comment]
@csoni111 my bad! will fix this ASAP.

**Comment by octacode** [comment]
Closing this for now. Will open a new PR to fix this issue

**Pull request closed without merging**